### ISSUE TITLE
Recipes not removed

### ISSUE THREAD

**Issue by CA2020** [legendblade/CraftingHarmonics#3]
I notice that the recipes get added with no problem, but I can't remove recipes.

For example.

`				{
                    __comment: "Remove String",
                    __result: "",
                    type: "remove",
                    what: "minecraft:string"
                },`

Isn't removing string recipes. Maybe I'm missing something here?

**Comment by legendblade** [comment]
Hrm.  That should definitely be working, unless I managed to break it in a recent release.  Let me test and get back to you.

**Comment by legendblade** [comment]
Ah, I think this might be related to a mod adding the recipe after CH tries to remove all of them.  Do you happen to know which mod is adding the recipe for string?  I'll use that to test that it's properly removing modded recipes.

Thanks.

**Comment by CA2020** [comment]
Sure, Quiet a few of them. For example.

TSON Craft, Reliquary, Modular System for example.

PS: I tried renaming CraftingHarmonic to have ZZZ in front. That helped in the past with similar mods, but not this time :P 

Thanks a lot.

**Comment by legendblade** [comment]
This should be resolved in [version 1.2.3](http://minecraft.curseforge.com/projects/crafting-harmonics/files/2294239), available now from Curse... as soon as it syncs across their network. :)

**Comment by CA2020** [comment]
Tried, but no succeed recipes I add show correctly, but it doesn't remove the ones that are registered by other mods.

**Comment by legendblade** [comment]
Huh.  Sorry about that.  It's strange: I added all three of the mods in that you mentioned, and was able to remove the leaves to string recipe that was added in by one of them with:

```javascript
{
    name: "Test file",
    description: "Testing for issue #3",
    sets: [
        {
            name: "default",
            operations: [
                {
                    type: "remove",
                    what: "minecraft:string"
                }
            ]
        }
    ]
}
```

At current, it won't be able to remove the modular systems recipes (the Modular / Electric Crushers), but, it did remove the other two that were included in testing.

**Comment by CA2020** [comment]
Ok Im trying on a new environment and it does work. I wonder if there is some cache or something on my other instance that keeps loading the recipe anyway. Gonna do some more testing ;)

PS: A file like this would be ok?
http://pastebin.com/kDSTtSnF
Or I need to separate sets?

**Comment by legendblade** [comment]
That's even weirder; the only 'cache' should be if you don't close and re-open the game between editing recipe sets (which is still a planned feature - just not sure when though), which you'd have to do to update the mod anyway, so, it shouldn't have been a caching issue?

For that matter, in the instance I was using to test this, I spelled "string" wrong the first few times (and spent extra time debugging it as a result), then, when I got it right, it started removing it.

Regarding having multiple recipes in the same file: yeah, that's perfectly fine.  My testing uses the sample file that gets generated when you first run the mod (more like: the sample is a copy-paste of my test script that I run through before most releases).
